File: TheComfortZone.DTO/User/UserInsertRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TheComfortZone.DTO.User
{
    public class UserInsertRequest
    {
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        public string FirstName { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        public string LastName { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(70)]
        public string Adress { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(15)]
        [Phone]
        public string PhoneNumber { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(30)]
        [EmailAddress]
        public string Email { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(20)]
        public string Username { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(20)]
        public string Password { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(20)]
        public string PasswordConfirmation { get; set; }
        [Required(AllowEmptyStrings = false)]
        [MaxLength(10)]
        public string RoleName { get; set; }
    }
}

File: the_comfort_zone_mobile/lib/model/user/user_insert_request.dart
import 'package:json_annotation/json_annotation.dart';
part 'user_insert_request.g.dart';

@JsonSerializable()
class UserInsertRequest {
  String? firstName;
  String? lastName;
  String? username;
  String? adress;
  String? phoneNumber;
  String? email;
  String? password;
  String? passwordConfirmation;
  String? roleName;

  UserInsertRequest();

  factory UserInsertRequest.fromJson(Map<String, dynamic> json) =>
      _$UserInsertRequestFromJson(json);

  /// Connect the generated [_$UserInsertRequestToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$UserInsertRequestToJson(this);
}
